packages/resources/src/App.ts

```diff
@@ -204,9 +206,25 @@ export class App extends cdk.App {
   setDefaultFunctionProps(
     props: FunctionProps | ((stack: cdk.Stack) => FunctionProps)
   ): void {
+    if (this.node.children.some((node) => node instanceof cdk.Stack))
+      throw new Error(
+        "Default function props for the app are already set. Please use app.addDefaultFunctionEnv or app.addDefaultFunctionPermissions to add more default properties"
+      );
```

Comment:
This is checking if `setDefaultFunctionProps()` is called after a Stack has been added. But the error message "Default function props for the app are already set" sounds like it's implying `setDefaultFunctionProps()` has been called multiple time?

Do we need 2 checks for this:
- if `setDefaultFunctionProps()` has been called => throw "Default function props for the app are already set..."
- if `this.node.children.some(...)` => throw "Cannot call setDefaultFunctionProps after a stack has been created"

Not sure if this makes sense?

Reply:
I didn't update this copy from when I was checking to see if they were calling it only once.  I think we can just keep the second copy - I'm not sure if preventing them from calling it multiple times is something we need to worry about.